base/gcutils.jl

```diff
@@ -49,29 +49,20 @@ Module with garbage collection utilities.
 """
 module GC
 
-# @enum-like structure
-struct CollectionType
-    x::Int
-end
-Base.cconvert(::Type{Cint}, collection::CollectionType) = Cint(collection.x)
-
-const Auto          = CollectionType(0)
-const Full          = CollectionType(1)
-const Incremental   = CollectionType(2)
-
 """
-    GC.gc(full::Bool=true)
-    GC.gc(collection::CollectionType)
+    GC.gc()
+    GC.gc(full::Bool)
 
-Perform garbage collection. The argument `full` determines whether a full, but more costly
-collection is performed. Otherwise, heuristics are used to determine which type of
-collection is needed. For exact control, pass an argument of type `CollectionType`.
+Perform garbage collection. The argument `full` determines the kind of collection: A full
+collection scans all objects, while an incremental collection only scans so-called young
+objects and is much quicker. If called without an argument, heuristics are used to determine
+which type of collection is needed.
 
 !!! warning
     Excessive use will likely lead to poor performance.
 """
-gc(full::Bool=true) = ccall(:jl_gc_collect, Cvoid, (Cint,), full)
-gc(collection::CollectionType) = ccall(:jl_gc_collect, Cvoid, (Cint,), collection)
+gc() = ccall(:jl_gc_collect, Cvoid, (Cint,), 0)
```

Comment:
Sorry, read this too quickly since it was supposed to be not a breaking change. This was a breaking change (as in CI has been broken for a week now because of this PR):
```suggestion
gc() = ccall(:jl_gc_collect, Cvoid, (Cint,), 1)
```



Please, please, please be careful about merging PRs that fail on CI (as this one did).

Reply:
Even if they are not wrapped in a type or user-facing, it might be good for this code to use locally-defined constants to make the code easier to read. My understanding is that these are the values:
```jl
const GC_AUTO = 0
const GC_FULL = 1
const GC_INCR = 2
```
Is that correct? If so, it would be good to define these constants and then use them in the method definitions.

Reply:
Crap, sorry for that, I had been too accustomed to win32 failing recently...

> Even if they are not wrapped in a type or user-facing, it might be good for this code to use locally-defined constants to make the code easier to read.

Agreed, I'll see about improving that next week.

Reply:
> Crap, sorry for that, I had been too accustomed to win32 failing recently...

:100: I wouldn't blame you here. CI needs to be fixed. The situation is atrocious.

Reply:
Yeah, I rather regret not merging #33741 to fix a major contributor months ago, but we really need to prioritize getting the other known fixes merged (#34336, #33785, and #34242 at least) too.